clients/snowflake/snowflake_dedupe_test.go

```diff
@@ -0,0 +1,74 @@
+package snowflake
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/artie-labs/transfer/clients/shared"
+	"github.com/artie-labs/transfer/lib/kafkalib"
+	"github.com/artie-labs/transfer/lib/stringutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func (s *SnowflakeTestSuite) TestDedupe() {
+	{
+		// Dedupe with one primary key + no `__artie_updated_at` flag.
+		tableID := NewTableIdentifier("db", "public", "customers")
+		stagingTableID := shared.TempTableID(tableID, strings.ToLower(stringutil.Random(5)))
+
+		parts := s.stageStore.generateDedupeQueries(tableID, stagingTableID, []string{"id"}, kafkalib.TopicConfig{})
+		assert.Len(s.T(), parts, 3)
+		assert.Equal(
+			s.T(),
+			fmt.Sprintf("CREATE OR REPLACE TRANSIENT TABLE %s AS (SELECT * FROM db.public.customers QUALIFY ROW_NUMBER() OVER (PARTITION BY by id ORDER BY id ASC) = 2)", stagingTableID.Table()),
```

Comment:
When you pull in the latest changes from `master` this will need to be `db.public."CUSTOMERS"` since we're always escaping and uppercasing Snowflake table names now.